File: C++17/백준/Silver/1969. DNA/DNA.cc
#include <iostream>
#include <vector>
#include <string>
#include <unordered_map>
#include <algorithm>

using namespace std;

int main() {
    int N, M;
    cin >> N >> M;
    vector<string> sequences(N);
    for (int i = 0; i < N; i++) {
        cin >> sequences[i];
    }
    string result = "";
    int distance_sum = 0;

    for (int i = 0; i < M; i++) {
        unordered_map<char, int> count;
        for (const auto& seq : sequences) {
            count[seq[i]]++;
        }
        char max_nucleotide = 'A';
        for (const auto& pair : count) {
            if (pair.second > count[max_nucleotide] || (pair.second == count[max_nucleotide] && pair.first < max_nucleotide)) {
                max_nucleotide = pair.first;
            }
        }
        result += max_nucleotide;
        distance_sum += N - count[max_nucleotide];
    }

    cout << result << endl;
    cout << distance_sum << endl;
    return 0;
}

File: Python/백준/Silver/1969. DNA/DNA.py
def hamming_distance(dna1, dna2):
    return sum(1 for a, b in zip(dna1, dna2) if a != b)

N, M = map(int, input().split())
sequences = [input().strip() for _ in range(N)]
result = ''
distance_sum = 0

for i in range(M):
    count = {'A': 0, 'C': 0, 'G': 0, 'T': 0}
    for seq in sequences:
        count[seq[i]] += 1
    max_nucleotide = max(count, key=lambda x: (count[x], -ord(x)))
    result += max_nucleotide
    distance_sum += N - count[max_nucleotide]

print(result)
print(distance_sum)